clang/lib/Driver/ToolChains/Gnu.cpp

```diff
@@ -546,6 +546,13 @@ void tools::gnutools::Linker::ConstructJob(Compilation &C, const JobAction &JA,
     ToolChain.addFastMathRuntimeIfAvailable(Args, CmdArgs);
   }
 
+  if (D.IsFlangMode()) {
+    CmdArgs.push_back("-lFortran_main");
```

Comment:
Not for this patch:
1. We will need an option to not include main. This is useful for creating libraries.
2. We might have to rename with Flang instead of Fortran.
3. We might need an option for pgmath.